[workloads] Add Current to the name of the workload manifest

steveisok opened the pull request:
https://github.com/dotnet/sdk/commit/f5cb7e3c3eb631d13a1bdc6033f399cf8a1ceadb introduced a change that would make `microsoft.net.workload.mono.toolchain` an outdated id in order to work around manifest errors when older previews of the .NET 7 SDK were installed. The unfortunate side effect of this change meant the name of the runtime manifest needs to be changed.

On the main branch, the manifest id will now contain current in the name for the .net version being developed.

Commit:
[workloads] Add Current to the name of the workload manifest

https://github.com/dotnet/sdk/commit/f5cb7e3c3eb631d13a1bdc6033f399cf8a1ceadb introduced a change that would make `microsoft.net.workload.mono.toolchain` an outdated id in order to work around manifest errors when older previews of the .NET 7 SDK were installed. The unfortunate side effect of this change means the name of the manifest needs to be changed.

On the main branch, the manifest id will now contain current in the name for the .net version being developed.

Commit:
Current was in the wrong spot. Moving it to the right place.

lewing reviewed (commented):
I think you'll have to update https://github.com/dotnet/runtime/blob/78f5a9839ac7d49a3cfaa6f4015a90b1cc846fa1/src/workloads/workloads.csproj#L127 or the msi name will be too long

Commit:
Add current to shorname

lewing commented:
I think we want to do the preview tagging before we do this?

Commit:
Merge remote-tracking branch 'upstream/main' into rename-current-manifest

steveisok commented:
Official build https://dev.azure.com/dnceng/internal/_build/results?buildId=2097223&view=results

Commit:
Add prerelease label to manifest name

Commit:
Add wildcard to support prerelease tagging

lewing commented:
@radical please take a look at what we need to do to make wbt happy

radical commented:
The generated package is `Microsoft.NET.Workload.Mono.ToolChain.Current.Manifest-8.0.100-alpha.1.8.0.0-ci.nupkg`, but we are trying to install `Microsoft.NET.Workload.Mono.ToolChain.Current.Manifest-8.0.100` package. I will try this out locally.

radical commented:
And the message says `      - Installing manifest: Microsoft.NET.Workload.Mono.ToolChain.Current/8.0.0-ci`, with the `-ci` tag. There is some mix up of the names.

Commit:
Merge remote-tracking branch 'upstream/main' into rename-current-manifest

Commit:
Alter emsdk dependency since the name changed

radical committed:
Use the sdkbandVersion with the prerelease suffix when installing workloads for testing

radical committed:
[wasm] Update emsdk manifest to alpha.1

radical committed:
InstallWorkloadFromArtifacts: Add workarounds for the manifest name

.. change.

The workload manifest has `.Current` added to the name now. This makes
the older name `microsoft.net.workload.mono.toolchain` invalid. But sdk
still installs this, which breaks the workload.

- As a workaround for the tests, remove manifest directory
- Also, update the name in the `IncludedWorkloadManifests.txt`

radical commented:
Needs updated packages from https://github.com/dotnet/emsdk/pull/276 .

steveisok commented on src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Current.Manifest/WorkloadManifest.json.in:
```diff
@@ -1,7 +1,7 @@
   "version": "${WorkloadVersion}",
   "depends-on": {
-    "Microsoft.NET.Workload.Emscripten.net7": "${EmscriptenVersion}"
+    "Microsoft.NET.Workload.Emscripten.Current": "${EmscriptenVersion}"
```
Does `$(SdkBandVersionForWorkload)` need to be part of the name?

steveisok replied:
Nevermind, manifest isn't part of the name here, so that doesn't make sense.

Commit:
Fix name in extends for wasm-tools

lewing commented:
we'll need to update the emsdk subscription in installer as well

lewing commented:
We're blocking installer now https://github.com/dotnet/installer/pull/15346#issuecomment-1406871800

radical committed:
[wasm] Bump emscripten manifest version to pick up the latest fixes

radical commented:
/azp run runtime

lewing commented:
installer followup https://github.com/dotnet/installer/pull/15366

lewing reviewed (approved):
We could update the emscripten version to latest but darc should take care of it

lewing commented:
I reran a couple of lanes but the failures were not related
